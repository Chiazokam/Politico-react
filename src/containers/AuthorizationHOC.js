import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { tokenExpired } from '../utils';

const token = localStorage.getItem('token');

const AuthorizationHOC = ({ component: Component, render, ...rest }) => (
  <Route
    {...rest}
    render={props => {
      if (token && !tokenExpired(token)) {
        return Component ? <Component {...props} /> : render(props)
      }
      return <Redirect to="/login" />
    }}
  />
);

export default AuthorizationHOC;
