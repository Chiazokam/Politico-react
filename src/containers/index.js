import Home from './home/Layout';
import About from './about/Layout';
import Signup from './signup/Layout';
import Login from './login/Layout';

export { Home, About, Signup, Login };
