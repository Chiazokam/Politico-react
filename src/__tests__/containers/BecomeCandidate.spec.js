import React from 'react';
import { shallow } from 'enzyme';
import { BecomeCandidate } from '../../containers';

describe('Become Candidate', () => {
  test('should render Become Candidate Container correctly', () => {
    const wrapper = shallow(<BecomeCandidate />);

    expect(wrapper).toMatchSnapshot();
  });
});
