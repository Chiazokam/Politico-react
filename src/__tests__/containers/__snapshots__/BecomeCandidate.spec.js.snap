// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Become Candidate should render Become Candidate Container correctly 1`] = `ShallowWrapper {}`;
